src/preprocess/encoder.ts

```diff
@@ -126,3 +132,38 @@ export class OneHotEncoder {
     return reshape(stack(cateTensors), [ -1 ]);
   }
 }
+
+export class LabelEncoder extends EncoderBase{
+  /**
+   * Encode a given feature into one-hot format
+   * @param x feature array need to encode
+   * @returns transformed one-hot feature
+   */
+  public async encode(x: Tensor | number[] | string[]): Promise<Tensor> {
+    if (!this.categories) {
+      throw TypeError('Please init encoder using init()');
+    }
+    const xTensor = checkArray(x, 'any', 1);
+    const xData = await xTensor.dataSync();
+    xTensor.dispose();
+    return tensor(xData.map((d: number|string) => this.cateMap[d]));
```

Comment:
```suggestion
    return tensor(xData.map((d) => this.cateMap[d]));
```
Is the type required?